src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php

```diff
@@ -241,7 +246,13 @@ public function update_item( $request ) {
 		} else {
 			$result = wp_insert_post( wp_slash( (array) $changes ), true );
 		}
+
 		if ( is_wp_error( $result ) ) {
+			if ( 'db_update_error' === $result->get_error_code() ) {
```

Comment:
Any chance this could come up because of an invalid input that e.g. violates some db constraint?

Reply:
Either way, if were get a database error, we should return that.

Reply:
@adamziel I'm not aware of any such cases, but if they do exist, we should probably update `wp_insert_post` to handle those errors to return something more descriptive.